app/assets/javascripts/report-a-problem.js

```diff
@@ -1,84 +1,48 @@
 (function() {
   "use strict";
-
   window.GOVUK = window.GOVUK || {};
 
-  window.GOVUK.reportAProblem = {
-    showErrorMessage: function (jqXHR) {
-      var response = "<h2>Sorry, we're unable to receive your message right now.</h2> " +
-                     "<p>We have other ways for you to provide feedback on the " +
-                     "<a href='/contact'>contact page</a>.</p>"
-      $('.report-a-problem-content').html(response);
-    },
-
-    promptUserToEnterValidData: function() {
-      GOVUK.reportAProblem.enableSubmitButton();
-      $('<p class="error-notification">Please enter details of what you were doing.</p>').insertAfter('.report-a-problem-container h2');
-    },
-
-    disableSubmitButton: function() {
-      $('.report-a-problem-container .button').attr("disabled", true);
-    },
-
-    enableSubmitButton: function() {
-      $('.report-a-problem-container .button').attr("disabled", false);
-    },
-
-    showConfirmation: function(data) {
-      $('.report-a-problem-content').html(data.message);
-    },
-
-    submit: function() {
-      $('.report-a-problem-container .error-notification').remove();
-      $('input#url').val(window.location);
+  var ReportAProblem = function ($container) {
+    this.$container = $container;
+    var $form = $container.find('form'),
+        form = new GOVUK.ReportAProblemForm($form);
+
+    this.addToggleLink($container);
+    $form.on('reportAProblemForm.success', this.showConfirmation.bind(this));
+    $form.on('reportAProblemForm.error', this.showError.bind(this));
+  };
+
+  ReportAProblem.prototype.addToggleLink = function($container) {
+    var $toggle = $('\
+      <div class="report-a-problem-toggle-wrapper js-footer">\
+        <p class="report-a-problem-toggle">\
+          <a href="">Is there anything wrong with this page?</a>\
+        </p>\
+      </div>');
+
+    $container.before($toggle);
+    $toggle.on("click", ".report-a-problem-toggle a", function(evt) {
+      $container.toggle();
+      evt.preventDefault();
+    });
```

Comment:
You can loose the argument to this method and make it work like an instance method using something like: 

```javascript
ReportAProblem.prototype.addToggleLink = function() {
  var $toggle = $('\
    <div class="report-a-problem-toggle-wrapper js-footer">\
      <p class="report-a-problem-toggle">\
        <a href="">Is there anything wrong with this page?</a>\
      </p>\
    </div>');

  this.$container.before($toggle);
  $toggle.on("click", ".report-a-problem-toggle a", function(evt) {
    this.$container.toggle();
    evt.preventDefault();
  }.bind(this));
}
```